FEAT/Added Job Management Page

dev13-suthar opened the pull request:
->Admin can visit the /manage page and see all list of active jobs, and Delete them As Well.
->From the Action button admin can edit and delete JOBS.
->Admin can Create Post. (redirects to /create Page).
->Fixed Typo in Footer
-> In this PR i have only done Delete Job Functionality which is deactivated because there is no session Management as Now.
Fixes:[#284]
![Screenshot 2024-09-05 183753](https://github.com/user-attachments/assets/2c295be5-7e55-4836-9555-d80ff11cb94d)

dev13-suthar commented:
@VineeTagarwaL-code

VineeTagarwaL-code commented:
Nice pr, can you fix the conflicts and give a video of your functioning

dev13-suthar commented:
ok

dev13-suthar commented:
https://github.com/user-attachments/assets/596359a2-9bf5-4a40-99a4-d09ef64cf124

dev13-suthar commented:
@VineeTagarwaL-code solved conflicts, Fixed UI bug( JOB was being displayed even after deleting until we perform hard reload). tag me once you review this.

VineeTagarwaL-code commented:
this looks great now, a good starting point can you one more feature where the admin can approve jobs will be really usefull

dev13-suthar commented:
Yes i can do that, but since there is no session logic, Signup,Register how can i determine logged in User is Admin or Not,? Why there is no signin/signup options?

VineeTagarwaL-code commented:
can you add that feature here in this PR itself, was gonna open a new one but if you can do it it will be great !

dev13-suthar commented:
sure i'll take this.
so i have to done signin,signup logic with nextAuth and then add job approval functionality. right?

VineeTagarwaL-code commented:
yeah it already there the code is not purged

dev13-suthar commented:
summary:
Added /api/auth/...nextAuth,
cleaned authOptions code so we can use session,
/manage admin only page,
admin can approve Jobs.
video: 

https://github.com/user-attachments/assets/53576361-de21-46be-8687-5016bfba5926

commit these?

VineeTagarwaL-code commented:
show auth flow with logout

dev13-suthar commented:
https://github.com/user-attachments/assets/72aec0fa-b123-4410-880b-76b4533d98ad

dev13-suthar commented:
We can raise issue for better login and logout button and signin/signup form

VineeTagarwaL-code commented:
comment out the login flow, only ahve for admin a lot of pr has this and they are more better on this tbh

dev13-suthar commented:
https://github.com/user-attachments/assets/1d8d4ea9-aa46-42bf-a584-0877cb12a883

ready to merge i think

dev13-suthar commented:
fixing

VineeTagarwaL-code commented:
build fails, fix please

dev13-suthar commented:
fixed

dev13-suthar commented:
it should work now

aakash2330 commented on src/components/JobManagementTable.tsx:
```diff
@@ -0,0 +1,205 @@
+
+const JobManagementTable = ({ jobs, searchParams, initialJobs }: props) => {
+  const { toast } = useToast();
+  const [DispJobs, setDispJobs] = useState<JobType[]>(initialJobs);
```
instead of just the initial jobs , can you make all of them server rendered ?

dev13-suthar replied:
it was server rendered before, but because of server renders when something gets change like if job approves then the datatable was not syncing with the DB unless u hard refresh

aakash2330 replied:
you'll have to revalidate the route , and then it'll work

aakash2330 commented on src/components/JobManagementTable.tsx:
```diff
@@ -0,0 +1,205 @@
+    }
+  };
+
+  const refreshJobs = async () => {
```
why do we need this function ? is it just to fetch the latest changes ?.

dev13-suthar replied:
it calls everytime searchparams changes to fetch and put latest job with local state variable

aakash2330 replied:
you don't need this if youre rendering table on server side

dev13-suthar replied:
let this merge,  i will update it later

aakash2330 commented on src/components/JobManagementTable.tsx:
```diff
@@ -0,0 +1,205 @@
+  const totalPages =
+    Math.ceil((jobs.additional?.totalJobs || 0) / JOBS_PER_PAGE) ||
+    DEFAULT_PAGE;
+  const currentPage = parseInt(searchParams.page?.toString()) || DEFAULT_PAGE;
```
why do we need the toString() here  , if you parse the search params correctly before passing it to this table component , we wont need it

aakash2330 commented on src/components/JobManagementTable.tsx:
```diff
@@ -0,0 +1,205 @@
+                <PaginationPreviousButton
+                  searchParams={searchParams}
+                  currentPage={currentPage}
+                  baseUrl={APP_PATHS.JOBS}
```
can you also remove the need for this to be passed , its not required

dev13-suthar replied:
Pagination not required?

aakash2330 replied:
no , we don't need the baseUrl to be paaed

dev13-suthar commented on src/actions/job.action.ts:
```diff
@@ -348,3 +366,102 @@ export const updateJob = withServerActionAsyncCatcher<
+  if (session.user.role === 'USER') {
+    throw new Error('Unauth Access');
+  }
+  const result = ApproveJobSchema.parse(data);
```
but other actions are using parse only, if i use safeparse there will be an other check for if(data.success)

aakash2330 replied:
yes , it's better to safe parse

dev13-suthar commented on src/actions/job.action.ts:
```diff
@@ -348,3 +366,102 @@ export const updateJob = withServerActionAsyncCatcher<
+});
+
+export const getAllJobsForAdmin = withSession<
+  //not the ideal action we should make getAllJobs() better so we can have option to display verified and unverified jobs by Inputs
```
what im thinking is, if user role is admin return all job whether they are verified or unverified and if role is user return only verified job/  should i go with this?

dev13-suthar commented:
@VineeTagarwaL-code done ready to merge new feat

dev13-suthar commented:
merge these before major updates that will be coming like user onboarding. @VineeTagarwaL-code

dev13-suthar commented:
@VineeTagarwaL-code solved all reviews bugs and now Table is also Server rendered.

VineeTagarwaL-code commented:
fix conflicts need to merge

dev13-suthar commented:
@VineeTagarwaL-code  done

VineeTagarwaL-code commented:
give vid ref of current status of the pr

VineeTagarwaL-code commented:
A new issue will open to improve the ui of this page and possible add more functionality but great work 


Make sure you post your pr getting merged on x
